./gradlew :micronaut-test-type-pollution:test fails

### Expected Behavior

branch 4.8.x

```
./gradlew :micronaut-test-type-pollution:test
Version catalog 'mnTest' can be automatically imported. You can remove it from settings.gradle(.kts) file.
Type-safe project accessors is an incubating feature.
Project accessors enabled, but root project name not explicitly set for 'buildSrc'. Checking out the project in different folders will impact the generated code and implicitly the buildscript classpath, breaking caching.

> Task :micronaut-test-type-pollution:test FAILED



SUCCESS: Executed 0 tests in 247ms


[Incubating] Problems report is available at: file:///Users/sdelamo/github/micronaut-projects/micronaut-test/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':micronaut-test-type-pollution:test'.
> Test discovery failed

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.14.2/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 1s

```

### Actual Behaviour

_No response_

### Steps To Reproduce

_No response_

### Environment Information

_No response_

### Example Application

_No response_

### Version

4.8.3

--- a/test-type-pollution/build.gradle.kts
+++ b/test-type-pollution/build.gradle.kts
@@ -1,3 +1,5 @@
+import io.micronaut.build.TestFramework
+
 plugins {
     id("io.micronaut.build.internal.micronaut-test-module")
 }
@@ -28,4 +30,5 @@ micronautBuild {
     binaryCompatibility {
         enabled.set(true)
     }
+    testFramework = TestFramework.JUNIT5
 }

